taipy/gui/gui.py

```diff
@@ -1116,10 +1116,15 @@ def __handle_ws_get_module_context(self, payload: t.Any):
                 self._bind_custom_page_variables(
                     self._get_page(str(payload.get("path")))._renderer, self._get_client_id()
                 )
+                # get metadata if there is one
+                metadata: t.Dict[str, t.Any] = {}
+                page = self._get_page(str(payload.get("path")))
+                if hasattr(page._renderer, "_metadata"):
+                    metadata = getattr(page._renderer, "_metadata", {})
                 self.__send_ws(
                     {
                         "type": _WsType.GET_MODULE_CONTEXT.value,
-                        "payload": {"data": mc},
+                        "payload": {"context": mc, "metadata": json.dumps(metadata, cls=_TaipyJsonEncoder)},
```

Comment:
"metadata": json.dumps(metadata, cls=_TaipyJsonEncoder)} if metadata else None

Reply:
metadata wouldnt be None since I set the default as an empty array

Reply:
both the declaration and the default value of the getattr

Reply:
an empty dict would be falsy in python :-)

Reply:
oh it is fine I want an empty dict anyway to ensure correct json parsing on the frontend side :D

Reply:
Alright I will go with you :D